### ISSUE TITLE
Can't use with --experimental-test-coverage --test

### ISSUE THREAD

**Issue by gjuchault** [privatenumber/tsx#437]
### Acknowledgements

- [X] I searched existing issues before opening this one to avoid duplicates
- [X] I understand this is not a place for seek help, but to report a bug
- [X] I understand that the bug must be proven first with a minimal reproduction
- [X] I will be polite, respectful, and considerate of people's time and effort

### Minimal reproduction URL

https://codesandbox.io/p/devbox/tsx-import-coverage-dkkgs3

I had to use codesandbox to have recent node version, you need local reproduction:

<details>
  <summary>package.json</summary>

```json
{
  "name": "nodejs-sandbox",
  "scripts": {
    "start": "npm run test-a || npm run test-b || npm run test-c || npm run test-d",
    "test-a": "tsx --experimental-test-coverage --test index.test.ts",
    "test-b": "NODE_OPTIONS='--import tsx' node --experimental-test-coverage --test index.test.ts",
    "test-c": "NODE_V8_COVERAGE=./coverage c8 -r html tsx --experimental-test-coverage --test index.test.ts",
    "test-d": "NODE_OPTIONS='--import tsx' NODE_V8_COVERAGE=./coverage c8 -r html node --experimental-test-coverage --test index.test.ts"
  },
  "devDependencies": {
    "c8": "^8.0.1",
    "tsx": "^4.6.2"
  }
}
```

</details>

<details>
  <summary>index.test.ts</summary>

```ts
import { describe, it } from "node:test";
import * as assert from "node:assert/strict";

describe("case", () => {
  it("works", () => {
    assert.equal(1, 1);
  });
});
```

</details>

### Version

4.6.2

### Node.js version

v20+

### Package manager

npm

### Operating system

Linux

### Problem & Expected behavior

I'm trying to get tsx to work with the native test runner, coverage included
It seems to be failing with the following warning:

```
ℹ Warning: Could not report code coverage. TypeError: Cannot read properties of undefined (reading 'line')
```

Also, c8 is not able to generate any report:

```
Unknown%
```

### Contributions

- [ ] I plan to open a pull request for this issue
- [ ] I plan to make a financial contribution to this project

**Comment by privatenumber** [comment]
Is this a duplicate of https://github.com/privatenumber/tsx/issues/433?

**Comment by jbergstroem** [comment]
I have tested this too and get the same issue. `c8` and `tsx@4.2.1` seems to be the only way to get proper code coverage working.

**Comment by gjuchault** [comment]
@privatenumber I tried the solution you suggested in #433 locally and ran `npm link` to try on my reproduction case. It still didn't run

**Comment by gjuchault** [comment]
Should be fixed by [Node 22](https://github.com/nodejs/node/pull/52505) since they merged [this](https://github.com/nodejs/node/pull/52010)

**Comment by SuperchupuDev** [comment]
updated to node 22 doesn't solve the issue for me
![image](https://github.com/privatenumber/tsx/assets/53496941/e8356d65-d192-452e-8108-7a3e123dbbea)

**Comment by gjuchault** [comment]
Indeed, I just tried as well since Node 22 was released and I couldn't get it to work

**Comment by khaosdoctor** [comment]
Node team has addressed that in https://github.com/nodejs/node/issues/52775 to remove the error, the coverage though is still useless.

One thing that we saw is that it probably has to do with the source map generation, but I haven't yet tested it

**Comment by privatenumber** [comment]
I left a comment in https://github.com/nodejs/node/issues/52775#issuecomment-2123811461, but tsx has source maps with V8 coverage support (released 5 days ago in [v4.10.4](https://github.com/privatenumber/tsx/releases/tag/v4.10.4)).

I tested the minimal reproduction, and using c8 without `--experimental-test-coverage` works:
```
$ npx c8 -r html tsx --test index.test.ts
▶ case
  ✔ works (0.338ms)
▶ case (0.748625ms)
ℹ tests 1
ℹ suites 1
ℹ pass 1
ℹ fail 0
ℹ cancelled 0
ℹ skipped 0
ℹ todo 0
ℹ duration_ms 203.282667
```
Verified coverage is working with:
```
$ open coverage/index.html
```

But this doesn't with `--experimental-test-coverage`:
```
$ npx c8 -r html tsx --experimental-test-coverage --test index.test.ts
▶ case
  ✔ works (0.342167ms)
▶ case (0.768417ms)
ℹ Warning: Could not report code coverage. TypeError: Cannot read properties of undefined (reading 'line')
ℹ tests 1
ℹ suites 1
ℹ pass 1
ℹ fail 0
ℹ cancelled 0
ℹ skipped 0
ℹ todo 0
ℹ duration_ms 138.662125
```

IMO this is an issue blocked by Node.js

**Comment by khaosdoctor** [comment]
Tried to use it in the new 4.11 version, but it still shows the same error to me. I don't know if we should move this to Node or keep it here

**Comment by SuperchupuDev** [comment]
works for me now using node 22.3.0, looks like it was a node issue fixed by nodejs/node#53315 which was just shipped

**Comment by privatenumber** [comment]
Thanks.

Confirmed working with Node v22.3.0:
```
➜  sandbox (1) npx c8 -r html tsx --experimental-test-coverage --test index.test.ts
▶ case
  ✔ works (0.363709ms)
▶ case (0.820125ms)
ℹ tests 1
ℹ suites 1
ℹ pass 1
ℹ fail 0
ℹ cancelled 0
ℹ skipped 0
ℹ todo 0
ℹ duration_ms 296.576208
ℹ start of coverage report
ℹ --------------------------------------------------------------
ℹ file          | line % | branch % | funcs % | uncovered lines
ℹ --------------------------------------------------------------
ℹ index.test.ts | 100.00 |   100.00 |  100.00 |
ℹ index.ts      |  33.33 |   100.00 |  100.00 | 2-3
ℹ --------------------------------------------------------------
ℹ all files     |  83.33 |   100.00 |  100.00 |
ℹ --------------------------------------------------------------
ℹ end of coverage report
```
